include/trackcpp/tpsa.h

```diff
@@ -464,12 +476,30 @@ void Tpsa<V,N,TYPE>::get_power(unsigned int idx_, unsigned int* power_) {
     }
 }
 
+template <unsigned int V, unsigned int N, typename TYPE>
+TYPE Tpsa<V,N,TYPE>::eval(const std::vector<TYPE>& val) const {
+    if (val.size() != V) {
+        std::cerr << "Invalid vector size in Tpsa::eval!" << std::endl;

```

Comment:
why not throw an error here?

Reply:
to keep the standard. we have now used the throw/catch mechanism yet in trackcpp. but we can create a new PR for this. I think it is better than sending all exceptions to stderr...

Reply:
I think I'm starting to use the throw mechanism in my PR, but I don't know the behavior when using it in python. Does swig intercepts this and raise a python exception?

Reply:
dunno

Reply:
I looked into it (asked ChatGPT) and yes, there is a way to configure SWIG to wrap C++ exceptions into python exceptions. I'm implementing it in my PR on the IDs pass methods. With that in mind, I think throwing exceptions is a good way to handle problems in C++.

Reply:
@fernandohds564 , can you commit a change here using exception instead of cerr ?

Reply:
Maybe we could leave this kind of change to another PR, what do you think? I think there is a lot of places where we would benefit from this type of change.